Products/CMFPlone/browser/navigation.py

```diff
@@ -146,11 +149,13 @@ def _get_url(item):
                 return (get_id(item), item.getRemoteUrl)
             return get_view_url(item)
 
+        context_path = '/'.join(context.getPhysicalPath())
+
         # now add the content to results
         for item in rawresult:
-            if not navigation_settings.show_excluded_items:
-                if item.exclude_from_nav:
-                    continue
+            if not context_path.startswith(item.getPath()) and item.exclude_from_nav:  # noqa: E501
```

Comment:
Same remark as in the plone.app.layout PR: first check `item.exclude_from_nav` for slightly better performance.

Reply:
man @mauritsvanrees you're faster than me 😉 ... done

Reply:
*(sorry for the 4 months delay)*

If this is about performance, getting Asko's default getattr branch in plone.dexterity would be much more efficient, though some adjustments have to be done, like here :)

Not all objects might provide an `exclude_from_nav` attribute, so a `getattr()` would be safer IMHO